sympy/core/relational.py

```diff
@@ -575,7 +581,7 @@ def _eval_simplify(self, **kwargs):
                 measure = kwargs['measure']
                 if measure(enew) <= kwargs['ratio'] * measure(e):
                     e = enew
-            except ValueError:
+            except (ValueError, TypeError):
```

Comment:
Why is TypeError being caught?

We should be very careful about catching generic exception types because it can hide bugs.

Reply:
When `Eq` is returned from `_eval_simplify` of `Relational` , `_eval_simplify` of `Equality` further tries to rewrite terms.

https://github.com/sympy/sympy/blob/d053c1d4c3ca5420851dcd9a30dfc0e7cee98a48/sympy/core/relational.py#L570

Here it tries to rewrite `Set` and throws `TypeError`.

Reply:
Instead of catching the `TypeError` here there should be a check that both sides are Expr before the call to rewrite.